aspnetcore/fundamentals/minimal-apis/bindStreamPipeReader/7.0-samples/PipeStreamToBackgroundQueue/Program.cs

```diff
@@ -0,0 +1,47 @@
+using System.Threading.Channels;
+using BackGroundQueueService;
+var builder = WebApplication.CreateBuilder(args);
+// Create a channel to send data to the background queue.
+builder.Services.AddSingleton<Channel<Stream>>((_) => Channel.CreateUnbounded<Stream>());
+
+// Create a background queue service.
+builder.Services.AddHostedService<BackGroundQueue>();
+var app = builder.Build();
+
+app.Use(async (context, next) =>
+{
+    await next();
+    var reader = new StreamReader(context.Request.Body);
```

Comment:
What's this for?

Reply:
Just in case you want to do some logging👀

Reply:
Lets remove it 😄

Reply:
Alright 😅

Reply:
done boss

Reply:
I've learnt some new things. Thanks again @davidfowl